server/src/main/java/org/cloudfoundry/identity/uaa/util/JwtTokenSignedByThisUAA.java

```diff
@@ -126,12 +128,12 @@ public JwtTokenSignedByThisUAA checkSignature() {
         return checkSignature(fetchSignatureVerifierFromToken(this.tokenJwt));
     }
 
-    public JwtTokenSignedByThisUAA checkSignature(SignatureVerifier verifier) {
+    public JwtTokenSignedByThisUAA checkSignature(Object verifier) {
```

Comment:
It would be better if we can have a base type that can represent verifiers instead of taking `Object` type here. I suppose that you considered that already?